evap/results/exporters.py

```diff
@@ -122,46 +134,49 @@ def filter_evaluations(semesters, evaluation_states, degrees, course_types, cont
+                    question_results: list[RatingResult] = questionnaire_result.question_results  # type: ignore[assignment]
                     if all(
                         not question_result.question.is_rating_question
                         or question_result.counts is None
                         or sum(question_result.counts) == 0
-                        for question_result in questionnaire_result.question_results
+                        for question_result in question_results
                     ):
                         continue
                     if (
                         not contributor
                         or contribution_result.contributor is None
                         or contribution_result.contributor == contributor
                     ):
-                        results.setdefault(questionnaire_result.questionnaire.id, []).extend(
-                            questionnaire_result.question_results
-                        )
+                        results.setdefault(questionnaire_result.questionnaire.id, []).extend(question_results)
                         used_questionnaires.add(questionnaire_result.questionnaire)
-            evaluation.course_evaluations_count = evaluation.course.evaluations.count()
-            if evaluation.course_evaluations_count > 1:
+            evaluation.course_evaluations_count = evaluation.course.evaluations.count()  # type: ignore[attr-defined]
+            if evaluation.course_evaluations_count > 1:  # type: ignore[attr-defined]
                 course_results_exist = True
                 weight_sum = sum(evaluation.weight for evaluation in evaluation.course.evaluations.all())
-                evaluation.weight_percentage = int((evaluation.weight / weight_sum) * 100)
-                evaluation.course.avg_grade = distribution_to_grade(
+                evaluation.weight_percentage = int((evaluation.weight / weight_sum) * 100)  # type: ignore[attr-defined]
+                evaluation.course.avg_grade = distribution_to_grade(  # type: ignore[attr-defined]
                     calculate_average_course_distribution(evaluation.course)
                 )
             evaluations_with_results.append((evaluation, results))
 
         evaluations_with_results.sort(
             key=lambda cr: (cr[0].course.semester.id, cr[0].course.type.order, cr[0].full_name)
         )
-        used_questionnaires = sorted(used_questionnaires)
 
-        return evaluations_with_results, used_questionnaires, course_results_exist
+        return evaluations_with_results, sorted(used_questionnaires), course_results_exist
 
     def write_headings_and_evaluation_info(
-        self, evaluations_with_results, semesters, contributor, degrees, course_types
-    ):
+        self,
+        evaluations_with_results: list[tuple[Evaluation, Any]],
+        semesters: QuerySet[Semester],
```

Comment:
I think we only require `Collection[Semester]` here?

Reply:
Mypy says `Collection[Semester]` is not iterabel. this seems a valid argument